This restores QT option. note QT will fully be removed someday.
https://github.com/rstudio/rstudio/commit/c5082c2b9081bd23757aa99458d04d1eb8e2dbb4
--- a/cmake/globals.cmake
+++ b/cmake/globals.cmake
@@ -120,10 +120,10 @@ endif()
 # platform specific default for targets
 if(NOT RSTUDIO_TARGET)
 
-   # for macOS pro builds, default to electron as otherwise
+   # for macOS pro builds, default to desktop as otherwise
    # we will try (and fail) to build Linux-only launcher pieces
    if(APPLE AND RSTUDIO_PRO_BUILD)
-      set(RSTUDIO_TARGET "Electron")
+      set(RSTUDIO_TARGET "Desktop")
    else()
       set(RSTUDIO_TARGET "Development")
    endif()
@@ -140,9 +140,17 @@ if(NOT DEFINED RSTUDIO_SERVER)
    endif()
 endif()
 
-if(NOT DEFINED RSTUDIO_ELECTRON)
-   if(RSTUDIO_TARGET STREQUAL "Development" OR RSTUDIO_TARGET STREQUAL "Electron")
+if(NOT DEFINED RSTUDIO_DESKTOP)
+   if(RSTUDIO_TARGET STREQUAL "Electron")
       set(RSTUDIO_ELECTRON TRUE)
+   elseif(LINUX AND UNAME_M STREQUAL aarch64)
+      message(STATUS "Desktop builds not currently supported on aarch64; disabling RSTUDIO_DESKTOP")
+      set(RSTUDIO_DESKTOP FALSE)
+   elseif(APPLE AND UNAME_M STREQUAL arm64)
+      message(STATUS "Desktop builds not currently supported on arm64; disabling RSTUDIO_DESKTOP")
+      set(RSTUDIO_DESKTOP FALSE)
+   elseif(RSTUDIO_TARGET STREQUAL "Development" OR RSTUDIO_TARGET STREQUAL "Desktop")
+      set(RSTUDIO_DESKTOP TRUE)
    endif()
 endif()
 
@@ -152,8 +160,8 @@ if(NOT DEFINED RSTUDIO_ELECTRON)
 endif()
 
 # override if requested
-if(RSTUDIO_NO_ELECTRON)
-   set(RSTUDIO_ELECTRON FALSE)
+if(RSTUDIO_NO_DESKTOP)
+   set(RSTUDIO_DESKTOP FALSE)
 endif()
 
 # set session32 if specified
@@ -350,7 +358,7 @@ if(NOT DEFINED CMAKE_INSTALL_PREFIX)
    if(APPLE)
       set(CMAKE_INSTALL_PREFIX "/Applications")
    elseif(UNIX)
-      if(RSTUDIO_ELECTRON)
+      if(RSTUDIO_DESKTOP OR RSTUDIO_ELECTRON)
          set(CMAKE_INSTALL_PREFIX "/usr/local/lib/rstudio")
       else()
          set(CMAKE_INSTALL_PREFIX "/usr/local/lib/rstudio-server")
--- a/package/linux/CMakeLists.txt
+++ b/package/linux/CMakeLists.txt
@@ -100,15 +100,25 @@ if(RSTUDIO_SERVER)
     list(APPEND RSTUDIO_DEBIAN_DEPENDS "libpq5")
   endif()
 
-elseif(RSTUDIO_ELECTRON)
+elseif(RSTUDIO_DESKTOP OR RSTUDIO_ELECTRON)
 
-  # debian control files (Electron)
-  set(DEBIAN_POSTINST postinst-electron.in)
-  set(DEBIAN_POSTRM postrm-electron.in)
+  if(RSTUDIO_DESKTOP)
+    # debian control files (Qt)
+    set(DEBIAN_POSTINST postinst-desktop.in)
+    set(DEBIAN_POSTRM postrm-desktop.in)
 
-  # rpm scripts (Electron)
-  set(RPM_POSTINST postinst-electron.sh.in)
-  set(RPM_POSTRM postrm-electron.sh.in)
+    # rpm scripts (Qt)
+    set(RPM_POSTINST postinst-desktop.sh.in)
+    set(RPM_POSTRM postrm-desktop.sh.in)
+  else()
+    # debian control files (Electron)
+    set(DEBIAN_POSTINST postinst-electron.in)
+    set(DEBIAN_POSTRM postrm-electron.in)
+
+    # rpm scripts (Electron)
+    set(RPM_POSTINST postinst-electron.sh.in)
+    set(RPM_POSTRM postrm-electron.sh.in)
+  endif()
 
   # rpm dependencies
   string(FIND "${RSTUDIO_PACKAGE_OS}" "SUSE" SUSE_POS)
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -624,6 +624,9 @@ endif()
 if(RSTUDIO_SERVER)
    message(STATUS "Configured to build SERVER")
 endif()
+if (RSTUDIO_DESKTOP)
+   message(STATUS "Configured to build DESKTOP")
+endif()
 if (RSTUDIO_ELECTRON)
    message(STATUS "Configured to build ELECTRON")
 endif()
@@ -699,7 +702,7 @@ else()
       endif()
 
       # add desktop subprojects if we aren't building in server only mode
-      if(RSTUDIO_ELECTRON)
+      if(RSTUDIO_DESKTOP OR RSTUDIO_ELECTRON)
          add_subdirectory(diagnostics)
          add_subdirectory(desktop)
       endif()
@@ -708,12 +711,25 @@ else()
       # processing which we do in desktop
       add_subdirectory(session)
 
-      # desktop conf file used for dev-mode for Electron desktop
-      if(RSTUDIO_ELECTRON OR RSTUDIO_DEVELOPMENT)
+      # desktop conf file used for dev-mode for both Qt desktop and Electron desktop
+      if(RSTUDIO_DESKTOP OR RSTUDIO_ELECTRON OR RSTUDIO_DEVELOPMENT)
          configure_file(conf/rdesktop-dev.conf ${CMAKE_CURRENT_BINARY_DIR}/conf/rdesktop-dev.conf)
       endif()
 
       # configure desktop files
+      if(RSTUDIO_DESKTOP)
+         if(NOT WIN32)
+            if(NOT APPLE)
+               configure_file(rstudio-dev.in ${CMAKE_CURRENT_BINARY_DIR}/rstudio-dev)
+            else()
+               configure_file(rstudio-mac.in ${CMAKE_CURRENT_BINARY_DIR}/rstudio-dev)
+            endif()
+         else()
+            configure_file(rstudio.bat.in ${CMAKE_CURRENT_BINARY_DIR}/rstudio.bat @ONLY)
+            configure_file(rstudio-tests.bat.in ${CMAKE_CURRENT_BINARY_DIR}/rstudio-tests.bat @ONLY)
+         endif()
+      endif()
+
       if(RSTUDIO_ELECTRON AND WIN32)
          configure_file(rstudio-tests.bat.in ${CMAKE_CURRENT_BINARY_DIR}/rstudio-tests.bat @ONLY)
       endif()
--- /dev/null
+++ b/src/cpp/rstudio-dev.in
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+#
+# rstudio-dev
+#
+# Copyright (C) 2022 by Posit Software, PBC
+#
+# Unless you have received this program directly from Posit Software pursuant
+# to the terms of a commercial license agreement with Posit Software, then
+# this program is licensed to you under the terms of version 3 of the
+# GNU Affero General Public License. This program is distributed WITHOUT
+# ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+# AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+#
+#
+
+QT_PLUGIN_PATH="@QT_PLUGIN_PATH@" \
+    LD_LIBRARY_PATH="@QT_LIBRARY_PATH@" \
+    RS_CRASH_HANDLER_PATH="@RSTUDIO_TOOLS_ROOT@/crashpad/crashpad/out/Default/crashpad_handler" \
+    desktop/rstudio "$@"
--- a/src/node/CMakeLists.txt
+++ b/src/node/CMakeLists.txt
@@ -15,7 +15,7 @@
 cmake_minimum_required(VERSION 3.6.3)
 
 # don't add electron for development mode (since faster to work
-# iteratively using "npm start" and so forth)
+# iteratively using "yarn start" and so forth)
 if(RSTUDIO_ELECTRON AND NOT RSTUDIO_DEVELOPMENT)
    add_subdirectory(desktop)
 endif()
